fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultDimensionConfig.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.dimension;
+
+import com.yahoo.bard.webservice.data.config.names.DimensionName;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+
+import java.util.LinkedHashSet;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * A Dimension instance holds all of the information needed to construct a Dimension.
+ */
+public class DefaultDimensionConfig implements DimensionConfig {
+
+    private final DimensionName apiName;
+    private final String physicalName;
+    private final String description;
+    private final String longName;
+    private final String category;
+    private final LinkedHashSet<DimensionField> fields;
+    private final LinkedHashSet<DimensionField> defaultDimensionFields;
+    private final KeyValueStore keyValueStore;
+    private final SearchProvider searchProvider;
+
+    /**
+     * Construct a DefaultDimensionConfig Instance from Dimension Name , dimension fields and  default dimension fields.
+     *
+     * @param apiName  The API Name is the external, end-user-facing name for the dimension.
+     * @param physicalName  The internal, physical name for the dimension.
+     * @param description  A description of the dimension and its meaning.
+     * @param longName  The Long Name is the external, end-user-facing long  name for the dimension.
+     * @param category  The Category is the external, end-user-facing category for the dimension.
+     * @param fields  The set of fields for this dimension.
+     * @param defaultDimensionFields  The default set of fields for this dimension to be shown in the response.
+     * @param keyValueStore  The key value store holding dimension row data.
+     * @param searchProvider  The search provider for field value lookups on this dimension.
+     */
+    public DefaultDimensionConfig(
```

Comment:
Since this includes a `KeyValueStore` and a `SearchProvider`, it's actually not the config for a _general_ dimension, but it's the config for a `KeyValueStoreDimension` (which happens to be Fili's default).

So, perhaps a better name is `DefaultKeyValueStoreDimensionConfig`?